Storage/Processor/Config.php

```diff
@@ -39,19 +41,27 @@ public function __construct(StorageManager $storageManager, string $processor, s
         $this->assetHash = $assetHash;
         $this->options = $this->resolve($options);
         $this->configHash = $configHash;
+        $this->setCacheKeyAndFilename();
+
+        unset($options[EmsFields::CONTENT_PUBLISHED_DATETIME_FIELD]); //the published date can't invalidate the cache as it'sbased on the config hash now.
+    }
+
+    private function setCacheKeyAndFilename()
+    {
+        $this->cacheKey = $this->assetHash . '_' . $this->configHash;
         $this->filename = null;
 
-        if (!empty($this->getFileNames())) {
-            foreach ($this->getFileNames() as $filename) {
-                if (is_file($filename)) {
-                    $this->filename = $filename;
-                    $this->assetHash = $this->storageManager->computeFileHash($filename);
-                    break;
-                }
-            }
+        if ($this->getFileNames() === null) {
+            return;
         }
 
-        unset($options[EmsFields::CONTENT_PUBLISHED_DATETIME_FIELD]); //the published date can't invalidate the cache as it'sbased on the config hash now.
+        foreach ($this->getFileNames() as $filename) {
+            if (is_file($filename)) {
+                $this->filename = $filename;
+                $this->cacheKey .= '_' . $this->storageManager->computeFileHash($filename);
```

Comment:
personally I prefer
`$this->cacheKey = sprintf('%s_%s, $this->cacheKey, $this->storageManager->computeFileHash($filename)');`

But not required, might be nice to discuss about a standard one day? (PR approved)